### PR TITLE
Fix for #632 Issue when creating variables with incorrect names.

### PR DESCRIPTION
**Pull request by fviale**
- store original variables when we are validating the variable form (BackupVariables). Any modification will only apply to these original variables and will not pile up (which happens when we change the variable name inside the form at each unsuccessful attempt).
 - when the form is valid, remove this BackupVariables attribute.

### CODE REVIEW

File: app/scripts/proactive/model/Job.js [hunk 1]
```diff
@@ -332,17 +334,23 @@ define(
         }
 
         if (this.attributes.hasOwnProperty('Variables')) {
-          var variables = this.attributes.Variables;
+          var variables;
+          if (this.attributes.hasOwnProperty('BackupVariables')) {
+            this.attributes.Variables = JSON.parse(JSON.stringify(this.attributes.BackupVariables));
+          } else {
+            this.attributes.BackupVariables = JSON.parse(JSON.stringify(this.attributes.Variables));
```
Comment (chloegugli) [inline app/scripts/proactive/model/Job.js]: Can you add comments saying why you need this.attributes.BackupVariables and what you're doing with it?

File: app/scripts/proactive/model/Job.js [hunk 2]
```diff
@@ -332,17 +334,23 @@ define(
...
+          }
+          variables = this.attributes.Variables;
           var index = -1
           for (var i = 0; i < variables.length; i++) {
             if (variables[i].Name == variable.Name) {
               index = i;
```
Comment (chloegugli) [inline app/scripts/proactive/model/Job.js]: we could maybe add a break here

File: app/scripts/proactive/model/Task.js [hunk 3]
```diff
@@ -562,17 +564,23 @@ define(
...
+                    if (this.attributes.hasOwnProperty('BackupVariables')) {
+                      this.attributes.Variables = JSON.parse(JSON.stringify(this.attributes.BackupVariables));
+                    } else {
+                      this.attributes.BackupVariables = JSON.parse(JSON.stringify(this.attributes.Variables));
+                    }
+                    variables = this.attributes.Variables;
                     var index = -1
                     for (var i = 0; i < variables.length; i++) {
                         if (variables[i].Name == variable.Name) {
                             index = i;
```
Comment (chloegugli) [inline app/scripts/proactive/model/Task.js]: Same comment about the break

File: app/scripts/proactive/model/Task.js [hunk 4]
```diff
@@ -562,17 +564,23 @@ define(
                     variable.Value = "";
                 }
                 if (this.attributes.hasOwnProperty('Variables')) {
-                    var variables = this.attributes.Variables;
+                    var variables;
+                    if (this.attributes.hasOwnProperty('BackupVariables')) {
+                      this.attributes.Variables = JSON.parse(JSON.stringify(this.attributes.BackupVariables));
```
Comment (chloegugli) [inline app/scripts/proactive/model/Task.js]: Same comment about BackupVariables

File: app/scripts/proactive/model/Job.js (hunk 1 above)
Comment (fviale) [reply app/scripts/proactive/model/Job.js]: done

File: app/scripts/proactive/model/Job.js (hunk 2 above)
Comment (fviale) [reply app/scripts/proactive/model/Job.js]: done

File: app/scripts/proactive/model/Task.js (hunk 3 above)
Comment (fviale) [reply app/scripts/proactive/model/Task.js]: done

File: app/scripts/proactive/model/Task.js (hunk 4 above)
Comment (fviale) [reply app/scripts/proactive/model/Task.js]: done

**Pull request merged**